Content.Server/Sprite/Components/RandomSpriteColorComponent.cs

```diff
@@ -1,46 +1,18 @@
 using System.Collections.Generic;
-using Robust.Server.GameObjects;
 using Robust.Shared.GameObjects;
-using Robust.Shared.IoC;
 using Robust.Shared.Maths;
-using Robust.Shared.Random;
 using Robust.Shared.Serialization.Manager.Attributes;
 
 namespace Content.Server.Sprite.Components
 {
-    [RegisterComponent]
-    public class RandomSpriteColorComponent : Component, IMapInit
+    [RegisterComponent, ComponentProtoName("RandomSpriteColor")]
+    public class RandomSpriteColorComponent : Component
     {
-        public override string Name => "RandomSpriteColor";
+        // This should handle random states + colors for layers.
+        // Saame with RandomSpriteState
```

Comment:
"Saame"

Also while random state allows you to specify a layer, the random color component only works for the 0-th layer.
Maybe add a layer int data-field.

Reply:
TBH I'd rather combine it into the 1 component that takes in the layers with the specified states + colors which is also useful for holoparasite but OOS.

Reply:
fair, I guess this PR can just stay limited to purely ECSing